src/Extensions/Script/Serialization/JavaScriptSerializer.cs

```diff
@@ -1,42 +1,244 @@
 // MIT License.
 
-using System.Collections.Specialized;
+using System.Diagnostics.CodeAnalysis;
 using System.Text;
+using System.Text.Json;
+using System.Web.Resources;
 
 namespace System.Web.Script.Serialization;
 
-// TODO: We'll want to use System.Text.Json and change it to be strongly typed
-internal sealed class JavaScriptSerializer
+public class JavaScriptSerializer
 {
-    public object MaxJsonLength { get; internal set; }
+    internal const string ServerTypeFieldName = "__type";
+    internal const int DefaultRecursionLimit = 100;
+    internal const int DefaultMaxJsonLength = 2097152;
 
-    internal static string SerializeInternal(OrderedDictionary orderedDictionary)
+    internal static string SerializeInternal(object o)
     {
-        throw new NotImplementedException();
+        JavaScriptSerializer serializer = new JavaScriptSerializer();
+        return serializer.Serialize(o);
     }
 
-    internal void Serialize(object value, StringBuilder builder, SerializationFormat javaScript)
+    internal static object Deserialize(JavaScriptSerializer serializer, string input, Type type, int depthLimit)
     {
-        throw new NotImplementedException();
+        ArgumentNullException.ThrowIfNull(input);
+
+        if (input.Length > serializer.MaxJsonLength)
+        {
+            throw new ArgumentException(AtlasWeb.JSON_MaxJsonLengthExceeded, nameof(input));
+        }
+
+        var serializeOptions = new JsonSerializerOptions
```

Comment:
The options should be saved in the JavaScriptSerializer as it caches things and would contain the registered converters